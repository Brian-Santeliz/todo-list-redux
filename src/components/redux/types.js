export const ADD_TASK = "ADD_TASK";
export const DELETE_TASK = "DELETE_TASK";

export const ADD_TASK_ERROR = "ADD_TASK_ERROR";
export const DELETE_TASK_ERROR = "DELETE_TASK_ERROR";
