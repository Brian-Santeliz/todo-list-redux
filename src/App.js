import Navbar from "./components/Navbar";
import Form from "./components/Form";
import TaskList from "./components/TaskList";
import { Container } from "./components/Container";
function App() {
  return (
    <>
      <Navbar />
      <Container>
        <Form />
        <TaskList />
      </Container>
    </>
  );
}

export default App;
