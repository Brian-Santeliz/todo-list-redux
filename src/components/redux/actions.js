import {
  ADD_TASK,
  DELETE_TASK,
  DELETE_TASK_ERROR,
  ADD_TASK_ERROR,
  UPDATE_TASK,
  UPDATE_TASK_ERROR,
} from "./types";

export const addTask = (task) => {
  if (!task) {
    return {
      type: ADD_TASK_ERROR,
    };
  }
  const payload = {
    task,
    id: Date.now(),
    state: false,
  };
  return {
    type: ADD_TASK,
    payload,
  };
};
export const deleteTask = (id) => {
  if (!id) {
    return {
      type: DELETE_TASK_ERROR,
    };
  }
  return {
    type: DELETE_TASK,
    payload: id,
  };
};

export const updateTask = (id) => {
  if (!id) {
    return {
      type: UPDATE_TASK_ERROR,
    };
  }
  return {
    type: UPDATE_TASK,
    payload: id,
  };
};
