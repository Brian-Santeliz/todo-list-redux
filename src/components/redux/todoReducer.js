import {
  ADD_TASK,
  ADD_TASK_ERROR,
  DELETE_TASK,
  DELETE_TASK_ERROR,
  UPDATE_TASK,
  UPDATE_TASK_ERROR,
} from "./types";
const initialState = {
  todos: JSON.parse(localStorage.getItem("todos")) || [],
  error: false,
};
export const todoReducer = (state = initialState, { type, payload }) => {
  switch (type) {
    case ADD_TASK:
      return {
        ...state,
        error: false,
        todos: [...state.todos, payload],
      };
    case DELETE_TASK:
      return {
        ...state,
        error: false,
        todos: state.todos.filter((todo) => todo.id !== payload),
      };
    case UPDATE_TASK:
      return {
        ...state,
        error: false,
        todos: state.todos.map((todo) =>
          todo.id === payload ? { ...todo, state: !todo.state } : todo
        ),
      };
    case ADD_TASK_ERROR:
    case DELETE_TASK_ERROR:
    case UPDATE_TASK_ERROR:
      return {
        ...state,
        error: true,
      };
    default:
      return state;
  }
};
