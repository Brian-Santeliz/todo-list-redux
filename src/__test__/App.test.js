import App from "../App";
import { render, screen, cleanup, waitFor } from "@testing-library/react";
import { Provider } from "react-redux";
import { store } from "../components/redux/store";
import userEvent from "@testing-library/user-event";
beforeAll(() =>
  render(
    <Provider store={store}>
      <App />
    </Provider>
  )
);
afterAll(() => cleanup);
test("Testing to <App /> component", async () => {
  const navbar = screen.getByTestId("navbar");
  const inputTask = screen.getByTestId("inputTask");
  const botonSave = screen.getByTestId("boton");
  const emptyMessage = screen.queryByTestId("empty-message");
  const list = screen.queryByTestId("list-item");

  expect(navbar).toBeInTheDocument();
  expect(navbar.textContent).toBe("Todo list Redux");
  expect(navbar.tagName).toBe("NAV");
  expect(emptyMessage).toBeInTheDocument();
  expect(emptyMessage.textContent).toBe("Don't have task");

  expect(list).not.toBeInTheDocument();
  expect(screen.queryByTestId("btn-delete")).not.toBeInTheDocument();
  expect(navbar.tagName).not.toBe("HEADER");
  expect(navbar.textContent).not.toBe("Todo List Redux");
  userEvent.type(inputTask, "Testing app with Jest & Testin-library");
  userEvent.click(botonSave);
  await waitFor(() => {
    expect(screen.queryByTestId("list-item")).toBeInTheDocument();
  });

  expect(emptyMessage).not.toBeInTheDocument();
  expect(emptyMessage).not.toBe("Don't have task");
  expect(screen.queryByTestId("list-item").tagName).toBe("LI");
  // expect(screen.queryByTestId("list-item")).toMatchSnapshot();

  userEvent.click(screen.queryByTestId("list-item"));
  expect(screen.queryByTestId("list-item")).toHaveStyle({
    cursor: "pointer",
    color: "#bdbbbb",
  });
  const btnEliminar = screen.queryByTestId("btn-delete");

  expect(btnEliminar.textContent).toBe("DELETE");
  expect(btnEliminar.tagName).toBe("BUTTON");
  expect(btnEliminar.className).toBe("btn btn-danger");

  userEvent.click(btnEliminar);
  expect(emptyMessage.textContent).toBe("Don't have task");
  expect(screen.queryByTestId("empty-message")).toBeInTheDocument();
  expect(list).not.toBeInTheDocument();
  expect(btnEliminar).not.toBeInTheDocument();
});
