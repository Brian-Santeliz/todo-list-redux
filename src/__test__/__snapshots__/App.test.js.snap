// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing to <App /> component 1`] = `
<li
  class="list-group-item d-flex justify-content-between align-items-center"
  data-testid="list-item"
  style="cursor: pointer; color: rgb(51, 51, 51);"
>
  <span>
    Testing app with Jest & Testin-library
  </span>
  <button
    class="btn btn-danger"
    data-testid="btn-delete"
  >
    DELETE
  </button>
</li>
`;
